tests/test_estimation.py

```diff
@@ -0,0 +1,41 @@
+import numpy as np
+import pymc3 as pm
+import theano.tensor as tt
+
+from pymc3_hmm.distributions import DiscreteMarkovChain, PoissonZeroProcess
+from pymc3_hmm.step_methods import FFBSStep
+from tests.utils import simulate_poiszero_hmm
+
+
+# the way we use this initial probability state vector
+def test_gamma_0_estimation():  # annotate this test with pytest
+
+    np.random.seed(2032)
+
+    poiszero_sim, _ = simulate_poiszero_hmm(
+        30, 150, pi_0=[0.5, 0.5]
+    )  # test [.01, .99], [.99, .01]
+    y_test = poiszero_sim["Y_t"]
+
+    with pm.Model() as test_model:
+        p_0_rv = np.array([0.5, 0.5])
+        p_1_rv = np.array([0.5, 0.5])
+
+        P_tt = tt.stack([p_0_rv, p_1_rv])  # transition matrix
+        P_rv = pm.Deterministic("P_tt", tt.shape_padleft(P_tt))
+
+        # how well is this estimated?
+        pi_0_rv = pm.Dirichlet("pi_0", np.r_[1, 1])  # "flat" prior for initial states
+
+        S_rv = DiscreteMarkovChain("S_t", P_rv, pi_0_rv, shape=y_test.shape[0])
+
+        PoissonZeroProcess("Y_t", 150, S_rv, observed=y_test)
+
+    with test_model:
+        ffbs = FFBSStep([S_rv])
+        trace = pm.sample()
```

Comment:
This `pm.sample` step raises this error:

```python
SamplingError: Initial evaluation of model at starting point failed!
```

And I'm having a hard time figuring out exactly what's causing that error, any suggestions on where to start debugging this?

Reply:
The initial point is available from `test_model.test_point`.  If you were to plug those values into the model's log-likelihood function (e.g. `test_model.logp(test_model.test_point)`) and get something like `-np.inf`, you would see an error like that.

From there, you can check the log-likelihoods for each term (e.g. `test_model.S_t.logp(test_model.test_point)`.

I'm guessing that the `S_t` values are causing the a `-np.inf` log-likelihood.  The zero/non-zero HMMs are _very_ unforgiving; if an observation is zero/non-zero and the corresponding state value in the initial value array is non-zero/zero, then you'll get `-np.inf`.  

In other words, if you set up a model that says there's a non-zero/zero value emitted from the zero/non-zero state's distribution, the log-likelihood is going to tell you that this situation is impossible (i.e. `-np.inf`).